### PR TITLE
fix custom pipelines

### PR DESCRIPTION
**Pull request by tinaun**

### CODE REVIEW

**Comment by vitvakatu** [comment]
Interesting solution...

File: src/render/mod.rs [hunk 1]
```diff
@@ -428,12 +428,13 @@ impl PipelineStates {
         })
     }
 
-    pub(crate) fn pso_by_material(
-        &self,
-        material: &Material,
-    ) -> &BasicPipelineState {
+    pub(crate) fn pso_by_material<'a>(
```
Comment (kvark) [inline src/render/mod.rs]: Are the explicit lifetimes really needed here? I'd expect them to default to `&self` anyway

Comment (tinaun) [reply src/render/mod.rs]: they are. the default inference is

```
fn pso_by_material<'a, 'b>(&'a self, &'b Material) -> &'a BasicPipelineState
```
with 'b being an independent lifetime which could be less than 'a. this forces them to have the same lifetime

Comment (tinaun) [reply src/render/mod.rs]: at least, thats what rustc told me

**Comment by kvark** [comment]
Alright, thank you!
bors r+

**Pull request merged**